### PR TITLE
Add configuration properties for custom resource attributes

### LINKED ISSUE #107: Configuration Properties for OTel Resource Attributes
Hi!

Currently we have a look at Sleuth OTel and it seems the only way to configure resource attributes is by providing a custom `Supplier<io.opentelemetry.sdk.resources.Resource>` like
```kotlin
@Configuration
class SpringCloudSleuthOtelConfiguration {
    @Bean
    fun otelResourceAttributes(): Supplier<Resource> = Supplier {
        Resource.builder()
            .put(ResourceAttributes.DEPLOYMENT_ENVIRONMENT, "local")
            .put(ResourceAttributes.SERVICE_VERSION, "1.0.0")
            .put(ResourceAttributes.SERVICE_NAMESPACE, "my-namespace")
            .build()
    }
}
```

It would be great if the starter could take care of that via configuration properties. I thought of something like
```yaml
spring:
  sleuth:
    otel:
      resource:
        deployment.environment: develop
        service.version: 1.0.0
        …
```

### PR DESCRIPTION
**Pull request by kschlesselmann**
Resolves #107

### CODE REVIEW

**Comment by kschlesselmann** [comment]
@marcingrzejszczak Maybe the `@ConditionalOnClass(ProcessResource.class)` on `OtelResourceConfiguration` is now a but much and should be moved to the respective `@Bean`s?

File: spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelResourceProperties.java [hunk 1]
```diff
@@ -34,6 +37,11 @@ public class OtelResourceProperties {
 	 */
 	private boolean enabled = true;
 
+	/**
+	 * Map of custom resource attributes (e.g. service.version)
+	 */
+	private Map<String, String> attributes = null;
```
Comment (marcingrzejszczak) [inline spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelResourceProperties.java:43]: we can always initialize this to an empty map - that way no NPE checks have to take place

File: spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelResourceProperties.java [hunk 2]
```diff
@@ -42,4 +50,12 @@ public void setEnabled(boolean enabled) {
 		this.enabled = enabled;
 	}
 
+	public Map<String, String> getAttributes() {
+		return attributes;
```
Comment (marcingrzejszczak) [inline spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelResourceProperties.java:54]: `return this.attributes`

**Review by marcingrzejszczak — COMMENTED** [review]
> @marcingrzejszczak Maybe the @ConditionalOnClass(ProcessResource.class) on OtelResourceConfiguration is now a but much and should be moved to the respective @Beans?

Yeah we should create a static class (`OtelProcessResourceConfiguration`?) inside the `OtelResourceConfiguration` that would have the `@ConditionalOnClass` on it.

**Comment by kschlesselmann** [comment]
@marcingrzejszczak I pushed some changes to resolve your findings. Hope everything is in order now 👍🏻

**Review by marcingrzejszczak — COMMENTED** [review]
Added @Configuration to the OtelProcessResourceConfiguration class

**Review by marcingrzejszczak — APPROVED** [review]

**Review by marcingrzejszczak — APPROVED** [review]
LGTM

**Comment by marcingrzejszczak** [comment]
I tried to modify the PR in web editor but apparently sth went wrong

**Pull request merged**